### PR TITLE
Add hydria-based charcoal water purifier

### PR DESCRIPTION
**Pull request by Tamiore**
#### Summary
Content "Add clay_hydria-based large charcoal water purifier."

#### Purpose of change

As is, charcoal water purifier required a _plastic_ jug. That's rather restrictive, given how literally any large waterproof container should do.

#### Describe the solution

Added a bigger  charcoal water purifier that uses clay hydria as a container.

#### Testing

Simple .json change, works fine locally

#### Additional context

There is no clay container of exactly one gallon, so I made the clay version bigger.

### CODE REVIEW

**Comment by Tamiore** [comment]
Added "NO_UNLOAD" to both charcoal purifiers. 
Because, presumably, the whole "crafting" of those involves packing charcoal in a rather specific manner that not easy to further manipulate without disassembling the whole thing.

File: data/json/items/tool/cooking.json [hunk 1]
```diff
@@ -125,15 +125,38 @@
...
-    "material": [ "wood", "plastic" ],
+    "material": [ "plastic" ],
     "symbol": ";",
     "color": "light_gray",
-    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "charcoal": 150 } } ],
-    "initial_charges": 150,
-    "max_charges": 150,
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "charcoal": 75 } } ],
+    "initial_charges": 75,
+    "max_charges": 75,
```
Comment (Qrox) [inline data/json/items/tool/cooking.json:133]: #50035 modifies the crafting recipe to require 150 charcoal instead.

Comment (Tamiore) [reply data/json/items/tool/cooking.json:133]: I've already modified the cooking item Item to have 75 charges.

At least I think I did...

**Comment by occulti** [comment]
Bugfixes #50005 too but with extra content so have closed my previous pull request

**Pull request merged**